src/visualiser/svg.js

```diff
@@ -0,0 +1,192 @@
+        cir.setAttribute("r", r);
+        cir.setAttribute("fill", color);
+        return cir;
+    }
+
+    /**
+     * @summary Creates and returns a svg text element
+     * @param {number} x the x position of the center of the text
+     * @param {number} y the y position of the center of the text
+     * @param {string} text the text to be displayed
+     * @param {string} fontSize the text to be displayed, default is determinded by svgLineSpace
+     * @returns {SVGTextElement} SVGTextElement with the text given as argument
+     */
+    function createText(x, y, text, fontSize = svgTextSize){
+        const textSvg = document.createElementNS("http://www.w3.org/2000/svg", "text");
+        textSvg.setAttribute("x", x);
+        textSvg.setAttribute("y", y);
+        textSvg.textContent = text;
+        textSvg.setAttribute("font-size", fontSize);
+        return textSvg;
+    }
+    // --------------------------------------------------- End of SVG constructors ----------------------------
+    /**
+     * @summary calculates the x coordinate of the circle's center from the LearningStyleValue(from -11 to 11) to a x coordinate on the line
+     * @param {number} LSvalue the LearningStyleValue form which to calculate the x value of the circle
+     * @returns {number} Returns the x coordinate of the circle
+     */
+    function circleXValue(LSvalue){
+        let xPos = svgLineSpace; // add the ofset from the left of the svg (the lines start some length inside the grey) this is equal to the -11 position
+        let percent = (LSvalue + 11) / RANGEWIDTH; // from -11 to 11, how many % is the position into the line? 0%=-11, 100%=11
+        let lineLength = svgWidth - (2 * svgLineSpace); // the length of the line
+        xPos += lineLength * percent; // how far into the line must the center of the circle be placed
+        return xPos;
+    }
+
+    /**
+     * @summary finds how many elements are close to each other and increases their size accordingly
+     * @param {Array} arrCircleSize the input array, this contains
+     * @returns {Array} Returns an array with the radius(size) of the circles to be made, so alle the circles can be seen
+     */
+    function closeby(arrCircleSize){ // something is wrong in this function, but it still kinda works
+        let A = new Array();
+        for (let i = 0; i < arrCircleSize.length; i++) {
+            A[i] = 0;
+            for (let j = i; j < arrCircleSize.length; j++) {
+                if (i !== j){
+                    if (range(arrCircleSize[i], arrCircleSize[j])){
+                        A[i]++;
+                        A[j]++;
+                    }
+                }
+            }
+            A[i]++; // add sig selv, så kan vi gange med det, ellers ganger vi med 0 og så kommer der problemer
+        }
+        // calc the circlesize in pixels by the size e.g. 1, 2, 3... depends on how many values are close to eachother
+        A.forEach((size, idx, arr) => {
```

Comment:
For loop ovenover og Array.forEach her?

Reply:
Og ingen dansk pls

Reply:
For loop fordi der er 2 i hinanden, dansk fixed, er array 4 each ok?

Reply:
lav et for of

Reply:
Done